### PR TITLE
[FRONT-73] Update UI to match current design

### PR DESCRIPTION
**Pull request by ArnoClaude**
### Description of Issue
[Linear Issue](https://linear.app/la-famiglia-project/issue/FRONT-73/create-edit-displayed-column)

- A modal has been added to edit columns.
- A new topbar has been implemented with several buttons; however, these buttons do not currently have associated functionality (only edit columns)
- A filter bar has been introduced, but it currently does not include any filters.
- The 'Owner' and 'Name' columns have been merged for better clarity.
- A new column for logos has been introduced, (actual images missing)
- Rows are now wrapped in a Link, leading to a detailed page.
- Styling modifications have been made across the platform, including:

1. Colored values based on thresholds.
2. Adjusted tag colors.
3. Reformatted header styles.
4. Tweaked styles in various components.
5. Adjusted font weights, capitalization, icon colors, and the border radius of rows.

### How I implemented
I used the [Tailwind UI Dropdown](https://tailwindui.com/components/application-ui/elements/dropdowns) component

### Other
"@headlessui/react" is used for the Popup

### CODE REVIEW

File: src/components/pure/RepositoryTable/index.tsx [hunk 1]
```diff
@@ -92,43 +108,191 @@ const browseListColumns = [
...
+              leaveTo="transform opacity-0 scale-95"
+            >
+              {/* eslint-disable-next-line tailwindcss/migration-from-tailwind-2 */}
+              <Menu.Items className="absolute right-0 z-10 mt-2 w-44 origin-top-right rounded-md bg-gray-700 shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none">
+                <div className="py-1">
+                  {browseListColumns.map((column, index) => {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    const text =
+                      typeof column.header === 'function' ? column.header() : column.header
```
Comment (cnk1dev) [inline src/components/pure/RepositoryTable/index.tsx:221]: Build fails because of this. Line 220 was split up into 2 by prettier, that's why @ts-ignore doesn't apply for 221 any more and causes the error. Regardless of that, it should be fixed, not ignored: column.header() requires an argument of type `HeaderContext<Repository, number>`

File: src/components/pure/RepositoryTable/index.tsx [hunk 2]
```diff
@@ -26,6 +32,8 @@ type Repository = {
   category: RepositoryCategories
 }
 
+const nullFunc = () => null
+
 const columnHelper = createColumnHelper<Repository>()
 const browseListColumns = [
```
Comment (cnk1dev) [inline src/components/pure/RepositoryTable/index.tsx:38]: Move `browseListColumns` into a separate `columns.tsx` file next to `index.tsx` and import it to reduce the size of the huge table component

**Review by cnk1dev — CHANGES_REQUESTED** [review]
Great work, a few small things still need to be fixed though.

File: src/components/pure/RepositoryTable/index.tsx (hunk 1 above)
Comment (DamianGrosz2) [reply src/components/pure/RepositoryTable/index.tsx:221]: I dont really understand what argument that could be

File: src/components/pure/RepositoryTable/index.tsx (hunk 2 above)
Comment (DamianGrosz2) [reply src/components/pure/RepositoryTable/index.tsx:38]: ok done

**Review by cnk1dev — APPROVED** [review]

**Review by DamianGrosz2 — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
     "react:update-react-imports": "npx react-codemod update-react-imports"
   },
   "dependencies": {
+    "@headlessui/react": "^1.7.14",
```
Comment (pczern) [inline package.json:19]: add why packages were added in PR description

File: src/components/page/details/Chart.tsx [hunk 4]
```diff
@@ -53,6 +53,7 @@ const CustomTooltip = ({ active, payload, label }: TooltipProps<string, string>)
 
 const Chart = ({ data }: ChartProps) => {
   const dataTypes = useMemo(() => data.map((d) => d.name), [data])
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:55]: do not useMemo without measuring using withProfiler

File: src/components/page/details/Chart.tsx [hunk 5]
```diff
@@ -126,10 +129,11 @@ const Chart = ({ data }: ChartProps) => {
                 toggleModal()
               }}
             />
+
             <Modal isOpen={isModalOpen} onClose={toggleModal}>
               {dataTypes.map((item) => (
                 <Button
-                  key={dataType}
+                  key={nanoid()}
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:136]: we don't need nanoid(); this id is not consistent!!!! Component gets rerendered more often

File: src/components/page/details/Chart.tsx [hunk 6]
```diff
@@ -149,10 +153,11 @@ const Chart = ({ data }: ChartProps) => {
                 setTimeframeModalOpen(true)
               }}
             />
+
             <Modal isOpen={timeframeModalOpen} onClose={() => setTimeframeModalOpen(false)}>
               {TimeframeOptions.map((option) => (
                 <Button
-                  key={option.value}
+                  key={nanoid()}
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:160]: same here

File: src/components/page/details/ProjectInformation.tsx [hunk 7]
```diff
@@ -1,20 +1,22 @@
 import { StarIcon } from '@primer/octicons-react'
-import Button from '../../pure/Button'
+import Button from '@/components/pure/Button'
 
 type ProjectInformationProps = {
-  image: string // todo change type
+  image: string // TODO: change type
```
Comment (pczern) [inline src/components/page/details/ProjectInformation.tsx:5]: `@TODO` capital case

File: src/components/page/repositoryTable/Table.tsx
Comment (pczern) [inline src/components/page/repositoryTable/Table.tsx:1]: rename to RepositoryTable.tsx

File: src/components/page/repositoryTable/TopBar.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  columns: Column<Repository, unknown>[]
+  nullFunc: () => void
+}
+
+const TopBar = ({ columns, nullFunc }: TopBarProps) => (
+  <div className="flex flex-row justify-between border-b border-gray-800 px-6 pb-3.5">
+    {/* Filter, Sort, Edit Columns buttons */}
+    <div className="flex flex-row gap-3">
+      <div className="inline-block">
+        <Button
```
Comment (pczern) [inline src/components/page/repositoryTable/TopBar.tsx:22]: put an array with attributes above TopBar component and render all three, we want to keep components small

File: src/components/page/repositoryTable/columns.tsx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+const columnHelper = createColumnHelper<Repository>()
+
+const columns = [
+  columnHelper.accessor('logo', {
+    header: 'Logo',
+    cell: () => (
+      <div className="h-[20px] w-[20px] rounded-[5px] bg-gray-700" />
+      // <img
```
Comment (pczern) [inline src/components/page/repositoryTable/columns.tsx:30]: remove comment

File: src/components/pure/Sidebar/Box/GithubStatItem.tsx [hunk 10]
```diff
@@ -8,42 +8,64 @@ type GithubStatItemProps = {
...
-    >
-      <div className="flex flex-row items-center justify-center gap-[10px]">
-        {Icon && <Icon className="h-[14px] w-[14px] text-gray-500" />}
-        {IconMetric}
-        <span className={`text-xs not-italic leading-3 text-gray-100 ${paddingOn ? 'w-6' : ''}`}>
-          {value}
-        </span>
-        {growth && <span className="text-xs not-italic leading-3 text-gray-500">{growth}</span>}
+  hoverOn,
+  greenValue,
```
Comment (pczern) [inline src/components/pure/Sidebar/Box/GithubStatItem.tsx:23]: As only one color is chosen use one prop, you can provide defaults via an enum

File: src/data/repositoriesMock.ts [hunk 11]
```diff
@@ -1,8 +1,9 @@
-import ProgrammingLanguages from '../constants/programmingLanguages'
-import RepositoryCategories from '../constants/repositoryCategories'
+import ProgrammingLanguages from '@/constants/programmingLanguages'
+import RepositoryCategories from '@/constants/repositoryCategories'
 
 const repositories = [
   {
+    logo: '',
```
Comment (pczern) [inline src/data/repositoriesMock.ts:6]: Make your code work with logo undefined?

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/components/page/details/Chart.tsx [hunk 12]
```diff
@@ -81,12 +83,14 @@ const Chart = ({ data }: ChartProps) => {
   useEffect(() => {
```
Comment (DogukanGun) [inline src/components/page/details/Chart.tsx:83]: Please add a description for this useEffect

File: src/components/page/details/Chart.tsx [hunk 13]
```diff
@@ -126,10 +129,11 @@ const Chart = ({ data }: ChartProps) => {
                 toggleModal()
               }}
             />
+
             <Modal isOpen={isModalOpen} onClose={toggleModal}>
               {dataTypes.map((item) => (
```
Comment (DogukanGun) [inline src/components/page/details/Chart.tsx:134]: Please check dataTypes is empty or not. If the list empty please show a message for the user.

File: src/components/page/details/ProjectInformation.tsx [hunk 14]
```diff
@@ -1,20 +1,22 @@
...
   name: string
   eli5: string
   tags: string[]
 }
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars -- TODO
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
```
Comment (DogukanGun) [inline src/components/page/details/ProjectInformation.tsx:11]: Please do not disable anything.

File: src/components/page/details/ProjectInformation.tsx [hunk 15]
```diff
@@ -1,20 +1,22 @@
...
 const ProjectInformation = ({ image, name, eli5, tags }: ProjectInformationProps) => (
```
Comment (DogukanGun) [inline src/components/page/details/ProjectInformation.tsx:12]: You should define type for the all values in props.

File: src/components/page/details/ProjectInformation.tsx [hunk 16]
```diff
@@ -1,20 +1,22 @@
...
   <div className="border-b border-gray-800 px-7 py-4">
     <div className="mb-4 flex flex-row items-center justify-between">
       <div className="flex flex-row items-center">
         <div className="mr-4 h-[30px] w-[30px] rounded-[5px] bg-gray-500" />
+
         <h1 className="mr-3 text-16 font-medium">{name}</h1>
+
         {tags.map((text) => (
```
Comment (DogukanGun) [inline src/components/page/details/ProjectInformation.tsx:20]: Please check tags is empty or not. If the list empty please show a message for the user.

File: src/components/page/repositoryTable/Filterbar.tsx [hunk 17]
```diff
@@ -0,0 +1,28 @@
+import Button from '@/components/pure/Button'
+import { AiOutlineCalendar } from 'react-icons/ai'
+
+const nullFunc = () => null
```
Comment (DogukanGun) [inline src/components/page/repositoryTable/Filterbar.tsx:4]: Could you explain why you created this function ?

File: src/components/page/repositoryTable/columns.tsx [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+        <p className="ml-1 text-14 font-bold">{`/ ${info.row.original.name}`}</p>
+      </div>
+    )
+  }),
+  columnHelper.accessor('starCount', {
+    header: 'Stars',
+    cell: (info) => (
+      <GitHubStatisticItem
+        Icon={AiOutlineStar}
+        paddingOn={false}
```
Comment (DogukanGun) [inline src/components/page/repositoryTable/columns.tsx:51]: I think that you dont need use brackets for constant values. I mean you coudl write "2000" instead of {2000}. Could you check this?

**Review by DogukanGun — CHANGES_REQUESTED** [review]

**Review by DogukanGun — CHANGES_REQUESTED** [review]
Please check the feedbacks from me and there are some not answered feedbacks.

File: src/components/page/details/Chart.tsx (hunk 4 above)
Comment (DamianGrosz2) [reply src/components/page/details/Chart.tsx:55]: Please, when providing feedback also include possible suggestions :)

File: src/components/page/repositoryTable/TopBar.tsx (hunk 8 above)
Comment (DamianGrosz2) [reply src/components/page/repositoryTable/TopBar.tsx:22]: I disagree. this makes the code less readable, also it's just 3 buttons

File: src/components/page/repositoryTable/Table.tsx
Comment (DamianGrosz2) [reply src/components/page/repositoryTable/Table.tsx:1]: Actually we have already a component called like that

File: src/components/page/details/ProjectInformation.tsx (hunk 14 above)
Comment (DamianGrosz2) [reply src/components/page/details/ProjectInformation.tsx:11]: Its a todo, as long as we dont have images

File: src/components/page/details/ProjectInformation.tsx (hunk 15 above)
Comment (DamianGrosz2) [reply src/components/page/details/ProjectInformation.tsx:12]: They are above?

File: src/components/page/repositoryTable/Filterbar.tsx (hunk 17 above)
Comment (DamianGrosz2) [reply src/components/page/repositoryTable/Filterbar.tsx:4]: Yes thats just as a dummy whilst the actual component is being worked on

File: src/components/page/repositoryTable/columns.tsx (hunk 18 above)
Comment (DamianGrosz2) [reply src/components/page/repositoryTable/columns.tsx:51]: No doesnt work

File: src/components/page/repositoryTable/TopBar.tsx (hunk 8 above)
Comment (cnk1dev) [reply src/components/page/repositoryTable/TopBar.tsx:22]: agree with @DamianGrosz2, also component is only 124 lines so it technically conforms to the max 140 line "rule"

File: src/components/page/repositoryTable/Table.tsx
Comment (cnk1dev) [reply src/components/page/repositoryTable/Table.tsx:1]: imo we should also consider renaming a few other components but would do that in another PR

File: src/components/page/details/Chart.tsx (hunk 4 above)
Comment (cnk1dev) [reply src/components/page/details/Chart.tsx]: removed useMemo

File: src/components/page/details/Chart.tsx (hunk 5 above)
Comment (cnk1dev) [reply src/components/page/details/Chart.tsx]: my bad, fixed

File: src/components/page/details/Chart.tsx (hunk 6 above)
Comment (cnk1dev) [reply src/components/page/details/Chart.tsx]: fixed

File: src/data/repositoriesMock.ts (hunk 11 above)
Comment (cnk1dev) [reply src/data/repositoriesMock.ts:6]: fixed

File: src/components/page/details/Chart.tsx (hunk 13 above)
Comment (cnk1dev) [reply src/components/page/details/Chart.tsx]: now displaying a fallback message for the entire chart if there's no data

File: src/components/page/repositoryTable/TopBar.tsx (hunk 8 above)
Comment (pczern) [reply src/components/page/repositoryTable/TopBar.tsx:21]: the limit is too high

File: src/components/page/details/ProjectInformation.tsx (hunk 7 above)
Comment (pczern) [reply src/components/page/details/ProjectInformation.tsx:5]: `@`

File: src/components/page/repositoryTable/columns.tsx (hunk 9 above)
Comment (pczern) [reply src/components/page/repositoryTable/columns.tsx:30]: still there

File: src/components/pure/Sidebar/Box/GithubStatItem.tsx (hunk 10 above)
Comment (pczern) [reply src/components/pure/Sidebar/Box/GithubStatItem.tsx:23]: why are all requests marked resolved?

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/components/pure/Sidebar/Box/GithubStatItem.tsx (hunk 10 above)
Comment (DamianGrosz2) [reply src/components/pure/Sidebar/Box/GithubStatItem.tsx:23]: I had done them and resolved the. But didn't push bc I had to go then Clemens did them.

File: src/components/pure/Sidebar/Box/GithubStatItem.tsx [hunk 19]
```diff
@@ -17,33 +25,52 @@ const GithubStatItem = ({
...
+  redValue,
+  largeGap
+}: GithubStatItemProps) => {
+  const valueAsNumber = Number(value)
```
Comment (pczern) [inline src/components/pure/Sidebar/Box/GithubStatItem.tsx:33]: what? why is value not Number prop? What is value?

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/components/page/repositoryTable/TopBar.tsx (hunk 8 above)
Comment (cnk1dev) [reply src/components/page/repositoryTable/TopBar.tsx:21]: ok I'll address that in a later PR with some general refactoring though

File: src/components/page/repositoryTable/columns.tsx (hunk 9 above)
Comment (cnk1dev) [reply src/components/page/repositoryTable/columns.tsx]: gone now

File: src/components/pure/Sidebar/Box/GithubStatItem.tsx (hunk 19 above)
Comment (cnk1dev) [reply src/components/pure/Sidebar/Box/GithubStatItem.tsx]: fixed

**Review by pczern — APPROVED** [review]

File: src/components/page/details/ProjectInformation.tsx (hunk 16 above)
Comment (DamianGrosz2) [reply src/components/page/details/ProjectInformation.tsx]: Done
